wgengine/netstack/netstack.go

```diff
@@ -362,48 +362,46 @@ func (ns *Impl) UpdateNetstackIPs(nm *netmap.NetworkMap) {
 		}
 	}
 
-	ipsToBeAdded := make(map[tcpip.AddressWithPrefix]bool)
+	ipsToBeAdded := make(map[netip.Prefix]bool)
 	for ipp := range newIPs {
 		if !oldIPs[ipp] {
 			ipsToBeAdded[ipp] = true
 		}
 	}
-	ipsToBeRemoved := make(map[tcpip.AddressWithPrefix]bool)
+	ipsToBeRemoved := make(map[netip.Prefix]bool)
 	for ip := range oldIPs {
 		if !newIPs[ip] {
 			ipsToBeRemoved[ip] = true
 		}
 	}
 	ns.mu.Lock()
 	for ip := range ns.connsOpenBySubnetIP {
-		ipp := tcpip.AddrFromSlice(ip.AsSlice()).WithPrefix()
-		delete(ipsToBeRemoved, ipp)
+		delete(ipsToBeRemoved, netip.PrefixFrom(ip, ip.BitLen()))
 	}
 	ns.mu.Unlock()
 
 	for ipp := range ipsToBeRemoved {
-		err := ns.ipstack.RemoveAddress(nicID, ipp.Address)
+		err := ns.ipstack.RemoveAddress(nicID, tcpip.AddrFromSlice(ipp.Addr().AsSlice()))
 		if err != nil {
 			ns.logf("netstack: could not deregister IP %s: %v", ipp, err)
 		} else {
 			ns.logf("[v2] netstack: deregistered IP %s", ipp)
 		}
 	}
 	for ipp := range ipsToBeAdded {
+		if !ipp.IsValid() {
+			ns.logf("netstack: skipping invalid IP %s", ipp)
```

Comment:
if it's not valid, what different forms might this stringify as that you're using %s?

Reply:
change to log explicitly as `p.Addr()/p.Bits()`